advanced-game-design-team/Artifacts/Raycast Collisions/README.md

```diff
@@ -0,0 +1,112 @@
+# Raycasted Point Collision
+
+![](./rays.gif)
+
+## Introduction
+
+This guide explores a 2D raycasting system designed for detecting collisions between rays emanating from a point (typically the mouse cursor) and rectangular obstacles. The system comprises several key components that work together to create an efficient collision detection mechanism.
+
+## Core Components
+
+### 1. Ray Generation
+
+The system generates rays from the mouse cursor position in a 360-degree circle. The number of rays is defined by `NUM_RAYS`, which is set to 360 in this implementation, effectively creating one ray per degree.
+
+```cpp
+struct ray {
+    point_2d origin;
+    vector_2d direction;
+};
+
+// In the main loop:
+point_2d mouse = mouse_position();
+std::vector<ray_hit> hits = cast_rays(obstacles, mouse, NUM_RAYS, MAX_DISTANCE);
+```
+
+### 2. Obstacle Generation
+
+Rectangular obstacles are randomly generated at the start of the program, creating a dynamic environment for ray collision:
```

Comment:
I'd like to see something like "At the end we'll get to how these are drawn on the screen." added, making the overall "narrative" a little more cohesive as we move from 1->5, while also _implicitly_ emphasizing that our computation and drawing are decoupled.